src/ShipEngine.php

```diff
@@ -47,8 +49,15 @@ class ShipEngine
      */
     public function __construct(string $apiKey, Address\FormatterInterface $addressFormatter)
     {
-        $this->requestFactory = new Api\RequestFactory($apiKey);
         $this->addressFactory = new Address\Factory($addressFormatter);
+        $this->client = new Client([
+            'base_uri' => "https://api.shipengine.com/v1/",
+            'headers' => [
+                'Content-Type' => 'application/json',
+                'API-Key'      => $apiKey,
+            ],
+            'debug'   => true
```

Comment:
Going to take this out when I am done with this branch. This will be my last commit.

Reply:
Alright, took it out. Throwing this to be ready for review.